lib/Doctrine/Common/Annotations/TokenParser.php

```diff
@@ -136,6 +140,52 @@ public function parseUseStatement()
         return $statements;
     }
 
+    private function parseUseStatementV8(): array
+    {
+        $groupRoot = '';
+        $class = '';
+        $alias = '';
+        $statements = [];
+        $explicitAlias = false;
+        while (($token = $this->next())) {
+            $isNameToken = $token[0] === T_STRING || $token[0] === T_NS_SEPARATOR;
+
+            if ($token[0] === T_NAME_QUALIFIED || $token[0] === T_NAME_FULLY_QUALIFIED) {
+                $class .= $token[1];
+
+                $classSplit = explode('\\', $token[1]);
+                $alias = $classSplit[count($classSplit) - 1];
+            } else if($token[0] === T_NS_SEPARATOR) {
+                $class .= '\\';
+            } else if ($token[0] === T_AS) {
+                $explicitAlias = true;
+                $alias = '';
+            } else if ($isNameToken && !$explicitAlias) {
```

Comment:
this should use `$token[0] === T_STRING` rather than `$isNameToken`, as the case of `$token[0] === T_NS_SEPARATOR` would never reach that case anyway.
Same for the next condition